pkg/cmd/build/build.go

```diff
@@ -271,6 +271,16 @@ func OptsFromBuildInfo(manifestName, svcName string, b *model.BuildInfo, o *type
 		}
 	}
 
+	if len(b.Secrets) > 0 {
+		if o.Secrets == nil {
+			o.Secrets = make([]string, 0)
```

Comment:
`o.Secrets = make([]string, 0, len(b.Secrets))`

Reply:
I will go forward with an empty slice with 0 length so it will increment when reading the struct key and value